engines/python/setup/djl_python/huggingface.py

```diff
@@ -208,75 +208,20 @@ def load_stopping_criteria_list(self, stop_sequence):
-        errors = {}
-        batch = inputs.get_batches()
-        for i, item in enumerate(batch):
-            try:
-                content_type = item.get_property("Content-Type")
-                input_map = decode(item, content_type)
-            except Exception as e:  # pylint: disable=broad-except
-                logging.warning(f"Parse input failed: {i}")
-                input_size.append(0)
-                errors[i] = str(e)
-                continue
-
-            if is_chat_completions_request(input_map):
-                _inputs, _param = parse_chat_completions_request(
-                    input_map,
-                    is_rolling_batch_enabled(self.hf_configs.rolling_batch),
-                    self.tokenizer)
-            else:
-                _inputs = input_map.pop("inputs", input_map)
-                _param = input_map.pop("parameters", {})
-
-            # Per request streaming is only supported by rolling batch
-            if is_rolling_batch_enabled(self.hf_configs.rolling_batch):
-                _param["stream"] = input_map.pop("stream", False)
-
-            if not isinstance(_inputs, list):
-                _inputs = [_inputs]
-            input_data.extend(_inputs)
-            input_size.append(len(_inputs))
-
-            if "cached_prompt" in input_map:
-                _param["cached_prompt"] = input_map.pop("cached_prompt")
-            if "seed" not in _param:
-                # set server provided seed if seed is not part of request
-                if item.contains_key("seed"):
-                    _param["seed"] = item.get_as_string(key="seed")
-            if "output_formatter" not in _param:
-                _param["output_formatter"] = self.hf_configs.output_formatter
-
-            for _ in range(input_size[i]):
-                parameters.append(_param)
-
-            adapters_per_item = self._fetch_adapters_from_input(
-                input_map, item)
-            if not isinstance(adapters_per_item, list):
-                adapters_per_item = [adapters_per_item]
-
-            if adapters_per_item:
-                found_adapters = True
-            else:
-                # inference with just base model.
-                adapters_per_item = [""] * len(_inputs)
-
-            adapters.extend(adapters_per_item)
-            if len(_inputs) != len(adapters_per_item):
-                logging.warning(
-                    f"Number of adapters is not equal to the number of inputs")
-                errors[
-                    i] = "Number of adapters is not equal to the number of inputs"
-
-        self.adapters = adapters if found_adapters else None
-
-        return input_data, input_size, parameters, errors, batch
+    def parse_input(
+            self, inputs: Input
+    ) -> tuple[list[str], list[int], list[dict], dict, list]:
+        is_rolling_batch = is_rolling_batch_enabled(
+            self.hf_configs.rolling_batch)
+        parsed_input = parse_input_with_client_batch(
```

Comment:
I wonder if we can do this during initialization rather than per request.

We should know at the initialization phase whether we're using dynamic batch or rolling batch, and can have separate methods/objects like DynamicBatchInputParser, RollingBatchInputParser for example.

I also wonder if we should change the signature to return the ParsedInput object here? That could be useful for users trying to override this implementation, where they don't have to worry about which components of ParseInput to return and that interface stays relatively consistent for them over time